missions/views/assignment.py

```diff
@@ -22,27 +40,24 @@ def get_success_url(self):
         return reverse('mission_detail', kwargs={'pk': self.object.mission.id})
 
 
-class UpdateView(LoginRequiredMixin, PermissionRequiredMixin, View):
-    initial = {'key': 'value'}
+class UpdateView(LoginRequiredMixin, PermissionRequiredMixin, SingleTableMixin, FilterView):
     template_name = 'missions/assignee_form.html'
     nav_item = 'missions'
     permission_required = 'missions.change_assignment'
-
-    def get(self, request, *args, **kwargs):
-        mission_id = kwargs.pop('mission__id')
-        assigned_users = User.objects.filter(assignments__mission__id=mission_id)
-        candidates = Profile.objects.exclude(user__in=assigned_users)
-        return render(request, self.template_name, {'candidates': candidates, 'view': self})
+    table_class = CandidatesTable
+    model = Profile
+    filterset_class = ProfileFilter
 
     def post(self, request, *args, **kwargs):
+
         assignment_id = kwargs.pop('assignment__id')
-        mission_id = kwargs.pop('mission__id')
         profile_id = request.POST['assignee']
-        user = get_object_or_404(User, profile__pk=profile_id)
-        assignment = Assignment.objects.get(pk=assignment_id)
-        assignment.user = user
-        assignment.save()
-        return redirect(reverse('mission_detail', kwargs={'pk': mission_id}))
+        if profile_id:
+            user = get_object_or_404(User, profile__pk=profile_id)
+            assignment = Assignment.objects.get(pk=assignment_id)
+            assignment.user = user
+            assignment.save()
```

Comment:
You could also write `Assignment.objects.filter(pk=assignment_id).update(user__pk=user.pk)`.

Reply:
I am not sure if this lets the program work correctly because we have a signal listening for saving assignments and the reset some values of the user got changed.

Reply:
This bypasses signals. Good to know.